vega/vega
# build: add deploy preview of the Vega editor

## Motivation

- Fix https://github.com/vega/vega/issues/3924

## Changes

- Set explicit yarn and node versions so that the build step can succeed in Cloudflare CI
- Move installing data out of the postinstall as it was causing the initial install to break. I moved it to a docs building step
- Add a build script for setting up a site deploy preview

## Testing

<img width="1464" alt="image" src="https://github.com/vega/vega/assets/9020979/62fed5ec-ff0d-45b2-9e23-08dc0d82241b">

- Visit the preview URL: https://cameron-yick-deploy-previews.vega-628.pages.dev/
- Check that console logs from both `vega` and a subpackage like `vega-view` are both visible. 
  - (I'll remove these statements before merging)
